src/client.ts

```diff
@@ -170,6 +170,32 @@ export class PluggyClient extends BaseApi {
     )
   }
 
+  /**
+   * Fetch all transactions from an account
+   * @param accountId The account id
+   * @returns {Transaction[]} an array of transactions
+   */
+  async fetchAllTransactions(accountId: string): Promise<Transaction[]> {
+    const { totalPages, results: firstPageResults } = await this.fetchTransactions(accountId)
+    if (totalPages === 1) {
+      // just one page return transactions
+      return firstPageResults
+    }
+
+    const transactions: Transaction[] = [...firstPageResults]
+
+    for (let i = 1; i <= totalPages; i++) {
+      if (i === 1) {
+        // first page already fetched
+        continue
+      }
+      const paginatedTransactions = await this.fetchTransactions(accountId, { page: i })
```

Comment:
rather than starting the loop with a condition we already know won't occur, why not just start the loop when we actually want to?
also, usually `while` is more readable than `for-i`.

```suggestion
    // first page already fetched
    let page = 1
    
    while(page < totalPages) {
      page++
      const paginatedTransactions = await this.fetchTransactions(accountId, { page: i })
```

Reply:
Good point, I will change that. 
Thinking about the statement used, I usually prefer the ´for-i´ loop when it's already known the times you want to iterate, but maybe you are right here!! 
Also, if we want actually to don't fetch the first page in your example we should initialize page as 2 like -> `let page = 2` since the first request the API retrieves the first page :)